src/azul/tdr.py

```diff
@@ -162,12 +181,50 @@ def add_entity(self, entity_key, entity_type: str, entity_row: JSON) -> None:
         self.metadata[entity_key] = json.loads(entity_row['content'])
 
 
+class TerraDataRepoException(HTTPError):
+
+    def __init__(self, response: urllib3.response.HTTPResponse):
+        super().__init__(response.geturl(), response.status, response.msg, response.headers, None)
+
+
+@lru_cache
+def http() -> AuthorizedHttp:
```

Comment:
This shouldn't be so public. I'd prefer it to be a method on the AzulTDRClient class.

Reply:
Then you can use our cached_property again.

Reply:
I kept it outside the class because much like `azul.dss.shared_credentials` I wanted it to available in cases where an actual client instance wasn't readily available (keep in mind a valid dataset/snapshot is needed at initialization). Also didn't want to cache on the `self` parameter. So, for now at least, I'm sticking with `lru_cache` and moving to a classmethod.

Reply:
I don't think you hit the nail on the head though. I've mentioned this before: when you add a static/class method that is not a factory, you are doing something wrong in 90% of the cases. In this case there are some needless duplication, lack of abstraction.

Read https://en.wikipedia.org/wiki/Information_hiding